app/models/event.rb

```diff
@@ -4,4 +4,24 @@ class Event < ApplicationRecord
   belongs_to :event_schema, optional: false
 
   validates :body, :raw_data, presence: true
+
+  before_save :valid_schema_event?
+
+  delegate :schema, to: :event_schema
+
+  def valid_schema_event?
```

Comment:
I was thinking about calling this method `valid_body?`, It's actually what is being validated.

Reply:
done!